include/aspect/particle/output/interface.h

```diff
@@ -86,10 +86,37 @@ namespace aspect
                                const double current_time) = 0;
 
           /**
+           * Returns the particle file name taking into account the file index as well as the mpi proccess.
+           */
+          virtual
+          const std::string
+          get_file_name ();
```

Comment:
what does the default implementation of this class do? If the base class doesn't know what to do, then mark the function `=0`.

Also, is the function supposed to change member variables? If not, please mark the function `const`.